### PR TITLE
Issue #273 Do not provision the VM with OpenShift

### LINKED ISSUE #273: Add  '--no-provision' flag to minishift start
When the flag is set, only the VM is started and OpenShift does not get provisioned yet. You can use provided Docker daemon for "other" use cases or you can pre-seed the daemon with images prior then provisioning OpenShift. The exact details need to be worked out.

### PR DESCRIPTION
**Pull request by gbraad**
Fixes #273 Add experimental support to not provision the VM.

### CODE REVIEW

File: cmd/minishift/cmd/config/config.go [hunk 1]
```diff
@@ -97,6 +97,9 @@ var (
 	HostFoldersAutoMount = createConfigSetting("hostfolders-automount", SetBool, nil, nil, true, nil)
 	HostFoldersSftpPort  = createConfigSetting("hostfolders-sftp-port", SetInt, []setFn{validations.IsValidPort}, nil, true, nil)
 
+	// No Provision
+	NoProvision = createConfigSetting("no-provision", SetBool, nil, nil, true, nil)
```
Comment (praveenkumar) [inline cmd/minishift/cmd/config/config.go]: :+1:

File: cmd/minishift/cmd/start.go [hunk 2]
```diff
@@ -578,6 +583,7 @@ func initStartFlags() *flag.FlagSet {
 	startFlagSet.AddFlag(nameServersFlag)
 
 	if minishiftConfig.EnableExperimental {
+		startFlagSet.Bool(configCmd.NoProvision.Name, false, "Do not provision the VM with OpenShift (experimental)")
```
Comment (praveenkumar) [inline cmd/minishift/cmd/start.go]: :+1:

**Comment by LalatenduMohanty** [comment]
@gbraad  The only thing I am not sure about it that, the `oc` binary gets downloaded (if not cached) for --no-provision. But without OpenShift cluster it does not have any value. So seems unnecessary to download the `oc` binary. Either we fix this in this PR or create an issue to fix this later. WDYT?

**Comment by praveenkumar** [comment]
> The only thing I am not sure about it that, the oc binary gets downloaded (if not cached) for --no-provision. But without OpenShift cluster it does not have any value. So seems unnecessary to download the oc binary.

@LalatenduMohanty  @gbraad It does have value if you start the vm without provision then also you can use `minishift docker-env` and `minishift oc-env` then use download oc binary to just start cluster as per your requirement. This actually help us also to verify if there is any issue when doing plain `oc cluster up`. I would like to have it as it is without removing capablity of downloading oc binary.

**Comment by LalatenduMohanty** [comment]
> This actually help us also to verify if there is any issue when doing plain oc cluster up. I would like to have it as it is without removing capablity of downloading oc binary.

Fair enough.

**Comment by LalatenduMohanty** [comment]
@gbraad Thanks.

**Pull request merged**